### PR TITLE
fix: clean the finalizer of mcs if the ExposureType transforms from CrossCluster to LoadBalancer

### PR DESCRIPTION
**Pull request by jwcesign**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
When update mcs ExposureType from CrossCluster to LoadBalancer, we need to remove the finalizer.
If not, the resource can't be deleted

**Which issue(s) this PR fixes**:
Fixes #none

**Special notes for your reviewer**:
none

**Does this PR introduce a user-facing change?**:

```release-note
`karmada-controller-manager`: clean the finalizer of mcs if the ExposureType transforms from CrossCluster to LoadBalancer.
```

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
/assign

File: pkg/controllers/multiclusterservice/mcs_controller.go [hunk 1]
```diff
@@ -241,6 +241,15 @@ func (c *MCSController) handleMultiClusterServiceCreateOrUpdate(mcs *networkingv
 		if err := c.retrieveMultiClusterService(mcs, providerClusters); err != nil {
 			return err
 		}
+		finalizersUpdated := controllerutil.RemoveFinalizer(mcs, util.MCSControllerFinalizer)
+		if finalizersUpdated {
+			err := c.Client.Update(context.Background(), mcs)
+			if err != nil {
+				klog.Errorf("Failed to update MultiClusterService(%s/%s) with finalizer:%v", mcs.Namespace, mcs.Name, err)
```
Comment (RainbowMango) [inline pkg/controllers/multiclusterservice/mcs_controller.go:248]:
```suggestion
				klog.Errorf("Failed to remove finalizer(%s) from MultiClusterService(%s/%s): %v", util.MCSControllerFinalizer, mcs.Namespace, mcs.Name, err)
```

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**